### PR TITLE
refactor: change config to single struct

### PR DESCRIPTION
**Pull request by driftluo**
**What this PR does / why we need it**:

This PR change metadata config to a single struct, this is a break change

**Which docs this PR relation**:

- prepare for https://github.com/axonweb3/axon/discussions/1317

### !!Breaking Change!!

**CI Description**

| CI Name                                   | Description                                                     |
| ----------------------------------------- | --------------------------------------------------------------- |
| *Chaos CI*                                | Test the liveness and robustness of Axon under terrible network condition    |
| *Cargo Clippy*                            | Run `cargo clippy --all --all-targets --all-features`      |
| *Coverage Test*                           | Get the unit test coverage report                             |
| *E2E Test*                                | Run end-to-end test to check interfaces                         |
| *Code Format*                             | Run `cargo +nightly fmt --all -- --check` and `cargo sort -gwc`     |
| *Web3 Compatible Test*                    | Test the Web3 compatibility of Axon                               |
| *v3 Core Test*                            | Run the compatibility tests provided by Uniswap V3             |
| *OCT 1-5 \| 6-10 \| 11 \| 12-15 \| 16-19* | Run the compatibility tests provided by OpenZeppelin           |

**CI Usage**

> Check the CI you want to run below, and then comment `/run-ci`.

**CI Switch**

- [x] Chaos CI
- [x] Cargo Clippy
- [ ] Coverage Test
- [x] E2E Tests
- [x] Code Format
- [x] Unit Tests
- [x] Web3 Compatible Tests
- [x] OCT 1-5 And 12-15
- [x] OCT 6-10
- [x] OCT 11
- [x] OCT 16-19
- [x] v3 Core Tests

### CODE REVIEW

File: core/run/src/tests.rs [hunk 1]
```diff
@@ -25,7 +25,7 @@ const CONFIG_FILE: &str = "config.toml";
 
 const GENESIS_HASH: &str = "0x2cc987996d5d26d18cb76dceb85d9b46e4f05f11ff331247225d983ec7a7b78f";
 const GENESIS_STATE_ROOT: &str =
-    "0x65f57a6a666e656de33ed68957e04b35b3fe1b35a90f6eafb6f283c907dc3d77";
+    "0x47bd35cdb8bd43da1d5ce85ad77fd68de4d654fc2677ac36bd3e18cfda0ff136";
```
Comment (chaoticlonghair) [inline core/run/src/tests.rs]: Mark this PR as a break change in PR description.

Comment (Flouse) [reply core/run/src/tests.rs]: +1

**Review by Flouse — CHANGES_REQUESTED** [review]
Please rebase main branch to resolve the conflicts.

And run CI again, thanks.

**Review by KaoImin — APPROVED** [review]

**Review by chaoticlonghair — APPROVED** [review]
I didn't understand the implementation details of this feature clearly.

Since there is no obvious issue in coding, let's merge it to avoid blocking the following part; then we can test the feature later.

**Review by Flouse — APPROVED** [review]

**Pull request merged**